src/Generator/EnumGenerator/Name.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Laminas\Code\Generator\EnumGenerator;
+
+use function strrpos;
+use function substr;
+
+/**
+ * @internal
+ *
+ * @psalm-immutable
+ */
+final class Name
+{
+    private string $name;
+
+    private ?string $namespace;
+
+    private function __construct(string $name, ?string $namespace)
+    {
+        $this->name      = $name;
+        $this->namespace = $namespace;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getNamespace(): ?string
+    {
+        return $this->namespace;
+    }
+
+    public static function fromName(string $name): self
```

Comment:
if it's deemed we must have a fully qualified class name, I'll adopt this suggestion.

Reply:
I just think its a bit confusing since a property `name` exists, one could assume that the named ctor only sets the name. And since the fqcn of a class without a namespace is the class name itself it makes at least sense to me. But if you disagree just ignore it.

Reply:
I agree with `fromFqcn()` or `fromFullyQualifiedClassName()`. A global Enum with no namespace is fully qualified already, and hence would have no nuamespace associated; namespaced Enums will be split into a namespace + name. The name of the method should reflect this.

Reply:
I'll adopt the longer option since it's more explicit.